src/main/resources/WebApp/src/app/component/main/main.component.ts

```diff
@@ -8,16 +8,111 @@ import {PostService} from '../../service/post/post.service';
   styleUrls: ['./main.component.scss']
 })
 export class MainComponent implements OnInit {
+  currentPage: number;
+
+  /*posts: PostDto[];*/
+  posts: PostDto[][]; /*first dimention for the page,and second dimention for the Post*/
 
-  posts: PostDto[];
   constructor(private postService: PostService) { }
 
   ngOnInit() {
+    /*
     this.postService.getAllPosts().subscribe(
       posts => {
         this.posts = posts;
       }
     );
+    */
+    this.currentPage = 0;
```

Comment:
@Arkarine Pagination that we want to have here is not to get list of Posts from BE splitted by the pages number. Angular should make the request to the backend asking to get posts with page number and page size. BE responses with posts, Angular shows them. I'd recommend to make BE implementation first (first you create API), and then make Angular consuming this API.